File: swat_server/analytics/com.tonybeltramelli.swat.server/utils/UMath.py
__author__ = 'Tony Beltramelli www.tonybeltramelli.com - 29/08/2015'


class UMath:

    @staticmethod
    def normalize(range_min, range_max, x, x_min, x_max):
        return range_min + (((x - x_min) * (range_max - range_min)) / (x_max - x_min))

    @staticmethod
    def get_frequency(sampling_rate_ms):
        return 1 / (sampling_rate_ms * 1e-06)

File: server/analytics/advanced/utils/UMath.lua
-- Author Tony Beltramelli www.tonybeltramelli.com - 13/11/2015

local UMath = {}

function UMath.normalize(rangeMin, rangeMax, x, xMin, xMax)
	return rangeMin + (((x - xMin) * (rangeMax - rangeMin)) * (1 / (xMax - xMin)))
end

function UMath.normalizeTensor(tensor)
	local min = torch.min(tensor, 1)[1]
	local max = torch.max(tensor, 1)[1]
	local t = tensor:storage()
	
	for i = 1, #t do
		t[i] = UMath.normalize(0.0, 1.0, t[i], min, max)
	end
	return torch.Tensor(t)
end

return UMath